atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/CassandraMultinodeEteTest.java

```diff
@@ -0,0 +1,237 @@
+        return CASSANDRA_NODES.get(index);
+    }
+
+    private static InetSocketAddress getCassandraThriftAddressFromPort(int port) {
+        DockerPort dockerPort = CASSANDRA_DOCKER_SETUP.hostNetworkedPort(port);
+        String hostname = dockerPort.getIp();
+        return new InetSocketAddress(hostname, dockerPort.getExternalPort());
+    }
+
+    static void killCassandraContainer(String containerName) {
+        Container container = CASSANDRA_DOCKER_SETUP.containers().container(containerName);
+        try {
+            container.kill();
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    void startCassandraContainer(String containerName) throws InterruptedException {
+        Container container = CASSANDRA_DOCKER_SETUP.containers().container(containerName);
+        try {
+            container.start();
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        waitForAllPorts(container);
+        //TODO: node should join the cassandra cluster
+        waitForNodetoolToConfirmStatus(container, "UN", 3);
+    }
+
+    private static void waitForAllPorts(Container container) {
+        Awaitility.await()
+                .atMost(50, TimeUnit.SECONDS)
+                .until(() -> container.areAllPortsOpen().succeeded());
+    }
+
+    private static Callable<Boolean> canCreateCassandraClient(Container container) {
+        final CassandraKeyValueServiceConfig config = ImmutableCassandraKeyValueServiceConfig.builder()
+                .addServers(getCassandraThriftAddressFromPort(THRIFT_PORT_NUMBER_1))
+                .addServers(getCassandraThriftAddressFromPort(THRIFT_PORT_NUMBER_2))
+                .addServers(getCassandraThriftAddressFromPort(THRIFT_PORT_NUMBER_3))
+                .keyspace("atlasdb")
+                .replicationFactor(3)
+                .credentials(ImmutableCassandraCredentialsConfig.builder()
+                        .username("cassandra")
+                        .password("cassandra")
+                        .build())
+                .build();
+        return () -> {
+                try {
+                    int externalPort = container.port(9160).getExternalPort();
+
+                    new CassandraClientFactory(getCassandraThriftAddressFromPort(externalPort), config).create();
+                    return true;
+                } catch (Exception e) {
+                    return false;
+                }
+            };
+    }
+
+    private void assertAddTodoTransactionWasSuccessful(TodoResource todoClient) {
+        Todo todo = ImmutableTodo.of("some unique TODO item with UUID=" + UUID.randomUUID());
+
+        todoClient.addTodo(todo);
+        List<Todo> todoList = todoClient.getTodoList();
+
+        assertThat(todoList, hasItem(todo));
+    }
+
+    private void waitForNodetoolToConfirmStatus(Container container, String status, int expectedNodeCount) {
```

Comment:
Sanity check (as I've heard of a problem bringing the container back to life): does this work in the happy case where all nodes stay up?